Added Get and Set operations with custom Converter definitions.

wingunder opened the pull request:
The GetCell, SetCell, GetColumn and SetColumn now take an
optional (last) parameter, specifying what Converter it should
use. This is a very handy way to override the default Converter, per
cell or column. One could now convert non-homogeneous data inside a
csv.

Full examples of usage can be found in test035.cpp.

All interfaces are backwards compatible, requiring no changes to
existing code.

The use of typeid(), for routing the conversions in the Converter
class, was removed and re-implemented, using template specialization.

d99kris commented:
Thanks for the PR! I've been kind of busy the last week, but I hope I'll get a chance to take a look in the next few days.

wingunder commented:
@d99kris,
No stress. I'll probably add some more commits to get rid of some compiler warnings, etc. Once you're satisfied, I'll squash-commit all into one commit.

d99kris commented:
In general I feel this is a very good PR, thanks a lot for submitting it! :)
I added two suggestions for const'ness (but optional - not really important). 

One thing though - could you help check if README.md section "Custom Data Conversion" needs to be updated? 
Also if you'd like some attribution, feel free to add a "Contributors" section to the README.md - perhaps just a bullet and your (github user)name/link (totally optional).

Once you're satisfied with the PR I'll pull the changes and generate the API doc and bump the version.

wingunder commented:
Thanks. I'll create an example for the `Custom Data Conversion` documentation and I think I'll rather make a `Contributors` file and reference it in the `README.md`.

I must admit that I didn't generate the doxygen API doc yet, as I was not sure if this monster-patch would be accepted. I'll work on that as well and document all the new members, etc. I'll hopefully have something ready by tomorrow.

wingunder commented:
Hi @d99kris,
I added an `AUTHORS` file. Could you please extend this file with the rest of the authors.
Uncrustify was run on all relevant files for this PR and I added another example into the `README.md`. Surprisingly enough, the `Custom Data Conversion` part that was in the `README.md` file had no changes, as that part of the interface stayed untouched. I did however add another example and hope that it will be useful.

Please let me know the PR is ok like this and if you want me to squash-commit it.
Regards

d99kris commented:
Hi @wingunder - thanks a lot, I think the PR looks very good!
To be honest I don't receive a lot of PRs so I'm not sure what is the best process for merging them, but yeah feel free to squash-commit it - I guess it will then have the correct author (you) vs. me if I were to squash and merge..(?)
Thanks!

wingunder commented:
Hi @d99kris,
>I'm not sure what is the best process for merging them, but yeah feel free to squash-commit it

No problem, I squash-committed it according to this HOWTO: https://github.com/wprig/wprig/wiki/How-to-squash-commits

The diff between origin/master and wingunder/custom_commit is now just one commit c4ce0b2.

d99kris commented:
Thanks for your contribution @wingunder
